movement/filesystem.c

```diff
@@ -166,6 +166,28 @@ bool filesystem_read_file(char *filename, char *buf, int32_t length) {
     return false;
 }
 
+bool filesystem_read_line(char *filename, char *buf, int32_t offset, int32_t length) {
+    memset(buf, 0, length);
+    int32_t file_size = filesystem_get_file_size(filename);
+    if (file_size > 0) {
+        int err = lfs_file_open(&lfs, &file, filename, LFS_O_RDONLY);
+        if (err < 0) return false;
+        err = lfs_file_seek(&lfs, &file, offset, LFS_SEEK_SET);
+        if (err < 0) return false;
+        err = lfs_file_read(&lfs, &file, buf, min(length, file_size - offset));
+        if (err < 0) return false;
+        for(int i = 0; i < length; i++) {
```

Comment:
Or:

```
char *end_of_line = strchr(buf, '\n');
if (end_of_line)
  *end_of_line = '\0';
```

I think. But if the function signature is changed to return the offset, then I guess there's more pay-off in doing our own loop.